corehq/apps/hqadmin/views/users.py

```diff
@@ -194,15 +194,24 @@ def _validate_user_access(self, user):
         return True
 
     def get(self, request, *args, **kwargs):
+        self.domain = request.GET.get('domain')
         full_username = request.GET.get('as', '')
+
         if not full_username or '@' not in full_username:
-            return HttpResponseBadRequest('Please specify a user using ?as=user@domain')
+            msg = 'Please specify a user using ?as=user@domain\nOr a web-user using ?as=email&domain=domain'
+            return HttpResponseBadRequest(msg)
+
+        self.user = CouchUser.get_by_username(full_username)
+        if self.user.is_web_user and not self.domain:
+            msg = 'Please specify domain for web-user using ?as=email&domain=domain'
+            return HttpResponseBadRequest(msg)
 
-        username, domain = full_username.split('@')
-        if not domain.endswith(settings.HQ_ACCOUNT_ROOT):
-            full_username = format_username(username, domain)
+        if not self.user.is_web_user or not self.user:
+            username, self.domain = full_username.split('@')
+            if not self.domain.endswith(settings.HQ_ACCOUNT_ROOT):
```

Comment:
Odd, I don't think I've ever looked closely at this before, but if I'm reading this right, this supports specifying mobile workers either as `username@domain.commcarehq.org` _or_ just `username@domain`.  I hadn't seen that latter format before.

However, this change would break with the first of those formats, as the domain would be `domain.commcarehq.org`.  I think you'll need to strip the second part before setting it to `self.domain`